flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -628,6 +658,8 @@ getFunctionType(mlir::Type resultType, llvm::ArrayRef<mlir::Value> arguments,
 // TODO find nicer type to string infra or move this in a mangling utility
 // mlir as Type::dump(ostream) methods but it may adds !
```

Comment:
Did you consider maybe extending Mangler.cpp to mangle type names up for this intrinsic use case?  (Ordinarily, there is no need for generating families of functions overloaded with types.)

Reply:
That sounds nicer to put it with the mangling utilities. Maybe `lib/Optimizer/Support/InternalNames.cpp` is the best fit since this does not need any front-end structure (Symbols, Scope...) and may be needed by the optimizer later if we move part of the intrinsic down the chain.
However, I do not really see it as a fit in the NameUniquer struct, would that be OK there as an its own function
`std::string mangleIntrinsicProcedure(name, mlir::FunctionType);` ?

Reply:
I did move it in `lib/Optimizer/Support/InternalNames.cpp` for now.